src/main/java/com/gft/component/ListAlgorithmWrapper.java

```diff
@@ -0,0 +1,22 @@
+package com.gft.component;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+import org.springframework.stereotype.Component;
+@Component
+public class ListAlgorithmWrapper {
+	public HashMap<String,SimpleMovingAverage> getSimpleMovingAverages(){
```

Comment:
Klasa powinna trzymać Mapę w polu i wypełniać ją tylko raz w konstruktorze (albo springowe lazy init jeśli chcesz pokombinować) i zwracana getterem. Powinna to być Mapa<String,PredictionAlgorithm>